tests/providers/apache/beam/operators/test_beam.py

```diff
@@ -286,7 +287,7 @@ def test_exec_dataflow_runner(self, gcs_hook, dataflow_hook_mock, beam_hook_mock
             "jobName": job_name,
             "stagingLocation": "gs://test/staging",
             "region": "us-central1",
-            "labels": {"foo": "bar", "airflow-version": TEST_VERSION},
+            "labels": {"foo": "bar"},
```

Comment:
This test failed without `xdist` even if it is the only one test case.

Reply:
So changing here is not a proper fix

Reply:
Or this is in the another place?

Reply:
Yeah tests cases broken completely. It is expects on the specific order, and have a side effect one after each other.

Single test in main failed
```console
pytest tests/providers/apache/beam/operators/test_beam.py::TestBeamRunJavaPipelineOperator::test_exec_dataflow_runne
...
============================================================================= 1 failed, 4 warnings in 2.34s ==============================================================================
```

Entire pass
```console
pytest tests/providers/apache/beam/operators/test_beam.py
...
============================================================================= 37 passed, 4 warnings in 2.20s =============================================================================
```

Reply:
It's correct fix.

The test failed because it should NOT get version label if you look at BeamRunJavaPipelineOperator - you will see that it does NOT set default labels in the constructor - they are only set by `BeamDataflowMixin`s  `__get_dataflow_pipeline_options` - in the `execute` method. But those defaults were updated by both PythonPipelineOperator and GoPilpelineOperator - in their constructors.  So the test should not expect the version (because this test does not run `execute' of Java operator.  However in most cases this test got "airflow-version" if any of the Python/Go operator tests were run before - because default options were modified by them.

Now - I have no idea if Python and Go should also NOT have the defaults in the constructor - it is likely, because they use the same mixin, however - I am not sure. And I decided to play safe and rather than remove setting the defaults I decided to get rid of the side effect by deepcopyng the dict before setting the defaults.

Reply:
And operators itself 🙄  E.g. this one will failed in case of XComArgs or similar

https://github.com/apache/airflow/blob/17cc4c911c17e1d18061cdf805d854a3fda66b15/airflow/providers/apache/beam/operators/beam.py#L336-L338

Reply:
Side effect happen because it use mutable as defaults in tests, so it should be rather move into the function-scoped fixture. But today is too late, so I will try to proper fix it later, we have the same problem in the past.